### ISSUE TITLE
Plans 2023 (NPM essential): Refactor grid component to export the individual parts (features table, comparison grid, etc.)

### ISSUE THREAD

**Issue by chriskmnds** [Automattic/wp-calypso#78266]
Blocked by https://github.com/Automattic/wp-calypso/issues/78472 (more or less, as WooExpress imports the component directly)

[PlanFeatures2023Grid](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plan-features-2023-grid/index.tsx) should be exporting individual components that can be put together on consuming end. This would greatly simplify complexity/logic and allow for better refactors and maintenance across better decoupled parts. 

The current form is fairly substandard right now. Between this and `PlansFeaturesMain` (another wrapper), the concept of "big", "wrapper", "do everything" components seems to be the spotlight, whilst "small", "isolated" parts should really be the underlying design/architectural consideration.

- Export from the index file: 
    - Comparison grid
    - Features table
    - Interval Toggle
- Update the consuming end (PlansFearturesMain) to put these together as needed.

While at it, consider the following (can/should be done in separate PRs):

- Does the grid need to [query for active promotions](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plan-features-2023-grid/index.tsx#L306)? (especially with package migration on the horizon)
- Convert the class component to a function component (no more "renderThis" stuff that can't even be subjected to hooks)

**Comment by ddc22** [comment]
I did some initial work on this and I was hoping that I am not treading on your work in the pipeline @chriskmnds 🤔 
As an initial tasks I was thinking about removing this.
> (no more "renderThis" stuff that can't even be subjected to hooks)

- https://github.com/Automattic/wp-calypso/pull/80784

**Comment by chriskmnds** [comment]
I will put a sense of **order** here, with some notes. what I hope will avoid having to redo things or retrofit, and to unblock other things from moving forward:

1. Split out the components into `FeaturesGrid`, `ComparisonGrid`, `SpotlightPlan`, etc.
    1. class components are fine, `renderFoo` parts fine
    2. should have a clearly defined set of properties that are shared, and some that are specific, with a single `gridPlans` object for all (no `gridPlansForComparisonGrid`, `gridPlansForFeaturesGrid`, etc.). 
        1. no need to refactor deeply the "HTML table or cards" layout used for features grid and spotlight, unless necessary.
            1. Use and reuse existing as much as possible. 
            2. I am hoping for a `FunkyHtmlTableOrCards` component to surface and be subject to refactoring in follow-ups.
        3. there may be some coupling needed to accommodate mobile view changing plans order when spotlight plan exists, something we can work around temporarily (either with a subscriber or passing the spotlight plans into the features grid as well)
    4. updates `plans-features-main.tsx` to put these together. more lifecycle methods and refs will migrate there.
3. Go deeper into the "HTML table or cards" layout and put things into internal and exported functions
    1. all the `renderFoo` parts getting refactored into function components internal/exported
    2. not meaning to consider a different implementation at this point, just clean up existing
5. More follow-ups to bring things home e.g. clean up remaining parts from `1.ii` above

> and unblock other things from moving forward

After `1` above, we can move into implementing the escape hatch in https://github.com/Automattic/wp-calypso/issues/78113, which is the next in the queue for me. Parts `2 & 3` can be done in parallel.

Also, concerning: 

> Does the grid need to [query for active promotions](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plan-features-2023-grid/index.tsx#L306)? (especially with package migration on the horizon)

There is a separate issue for this.

@southp @jdc91 (please comment if you have a different sense of priorities from findings in the work you've done above 🎉)

**Comment by southp** [comment]
Thanks for outlining the path forward. The order makes sense to me. Speaking of other priorities, I'm wondering how you see the handed-out tasks from @aneeshd16 fit into this queue. i.e.

* [Add sticky behavior to the plan comparison grid header.](https://github.com/Automattic/wp-calypso/pull/80019)
* [Plans: Toggle layout based on screen size](https://github.com/Automattic/wp-calypso/pull/79129)
* [Refactor media queries in PlanFeatures2023Grid component](https://github.com/Automattic/wp-calypso/pull/79405)

Where on the roadmap would be the better timing to carry them forward?

**Comment by chriskmnds** [comment]
> Where on the roadmap would be the better timing to carry them forward?

This is a good question. I think right after first iteration here we can carry on with the last two refactors (clean up media queries). Probably not before, as some things may end up  moving into `plans-features-main`. The sticky behavior I'm not too sure, if it will involve different mobile/desktop, then maybe after the media refactors, but someone can start exploring already.

@southp

**Comment by southp** [comment]
Cool. Thanks for the clarification :) 
I'll make a note in them so whoever bumps into them will be aware.

**Comment by chriskmnds** [comment]
So this is now part of [Plans NPM: Layer out and export the individual grids](https://github.com/Automattic/wp-calypso/milestone/455) 

One base PR is in progress and waiting rebase and reviewing: #80915
Two follow-up issues waiting (what I think @jdc91 will take on?): #81078, #81076 (lesser priority)

**Comment by chriskmnds** [comment]
We can close this issue. The grids are exported individually, and separate issues exist in [Milestone](https://github.com/Automattic/wp-calypso/milestone/455) for the remaining parts/things to do.
